src/ros_filter.cpp

```diff
@@ -1916,6 +1916,15 @@ namespace RobotLocalization
       {
         ROS_ERROR_STREAM("Critical Error, NaNs were detected in the output state of the filter." <<
               " This was likely due to poorly conditioned process, noise, or sensor covariances.");
+        if(last_odom.header.stamp.sec > 0.0)
+        {
+          ROS_ERROR_STREAM("Apply Previous Pose");
+          geometry_msgs::PoseWithCovarianceStamped last_pose;
+          last_pose.header = last_odom.header;
+          last_pose.pose = last_odom.pose;
+          setPoseCallback(boost::make_shared<geometry_msgs::PoseWithCovarianceStamped>(last_pose));
+          // we might want to early return
```

Comment:
Hmm yeah, I think that's a good idea

Reply:
What's the upside/downside with early return versus publishing?

Reply:
it likely doesn't matter as far as I read through the code.
but it is safer to do early return to fit the original workflow: detecting nan, reset the pose and resume in next update cycle

Reply:
and both pass the integration tests